packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -288,6 +266,28 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
           Assertions.expectWithinTolerance(amountPaidByOperator, totalOperatorFees, TOLERANCE);
         });
 
+        it('Should preemtively check the rate limit before submitting EthereumTransaction', async function () {
+          process.env.HBAR_RATE_LIMIT_PREEMTIVE_CHECK = 'true';
+          try {
+            for (let i = 0; i < 50; i++) {
+              const largeContract = await Utils.deployContract(
+                largeContractJson.abi,
+                largeContractJson.bytecode,
+                accounts[0].wallet,
+              );
+              await largeContract.waitForDeployment();
+            }
+            expect.fail('The rate limiter will throw HBAR_RATE_LIMIT_PREEMTIVE_EXCEEDED error');
+          } catch (e) {
+            console.log(`asdaksjdhuqwiyeiqwuyhdkajshdkajhsiduuqywe`);
+            console.log(e);
+
+            expect(e.message).to.contain(predefined.HBAR_RATE_LIMIT_PREEMTIVE_EXCEEDED.message);
+          }
+
+          delete process.env.HBAR_RATE_LIMIT_PREEMTIVE_CHECK;
```

Comment:
Please, don't do this.

First of all, we shouldn't delete this env variable, but restore it to its original state.

Also, such tests should be wrapped inside a `describe` with the pre-conditions done in `beforeAll`/`beforeEach` and then we should make sure that the original env state is always restored by placing it inside `afterAll`/`afterEach`. 

If we don't do this inside `beforeEach`/`afterEach`/`beforeAll`/`afterAll` and keep this overriding and restoring of env vars inside individual tests:
- ⚠️ any assertions that fail during this test would skip the remaining piece of code after them that restores the env variable and possibly propagate an error to all other tests which expect this env variable to be in its original state.
- ℹ️ introduce unnecessary duplication in common tests

Example of a proper test setup:
```typescript
describe('given HBAR_RATE_LIMIT_PREEMTIVE_CHECK is enabled', () => {
  let hbarRateLimitPreemptiveCheck: string | undefined;
  
  beforeAll(() => {
    hbarRateLimitPreemptiveCheck = process.env.HBAR_RATE_LIMIT_PREEMTIVE_CHECK;
    process.env.HBAR_RATE_LIMIT_PREEMTIVE_CHECK = 'true';
  });
  
  afterAll(() => {
    process.env.HBAR_RATE_LIMIT_PREEMTIVE_CHECK = hbarRateLimitPreemptiveCheck;
  });
  
  it('Should preemtively check the rate limit before submitting EthereumTransaction', () => {
    ...
  });

  it('<Name of some other test which needs this env variable set to true>', () => {
    ...
  });
});
```

Reply:
Good point! Thanks for the suggestion!

Reply:
Still not resolved

Reply:
Hmm strange I couldve sworn i updated this 😅 thanks for pointing it out!

Reply:
Updated with the suggested logic. Thanks!